### DISCUSSION TITLE
Wix V4 or V5 - .NET framework 4.x install with msi

### DISCUSSION THREAD

**Discussion by sampathy10** [wixtoolset/issues#8670] [Questions]
I built msi installer using Wix V5. Now i am trying to install downloaded .net framework 4.8 runtime in the target machine if it is not available. 
I tried to use bundle, but the installer GUI is belongs to bundle but not msi setup. I wanted to use MSI install GUI as primary and install .net runtime as dependent. I could not find enough documentation online to achieve this. Sample working code for my scenario using Wix 5 will be great help.

**Comment by bevanweiss** [comment]
The Windows Installer system has some locks around MSI installers.  Only one is supposed to be active at a time, so using the MSI to install something which is potentially itself an MSI isn't advisable.

The Bundle is the right way.
https://wixtoolset.org/docs/tools/wixext/dotnet/

You should just be able to have another wxs file for your bundle, with appropriate .NET reference for the version you want to install in the chain
```xml
... stuff for the bundle description, upgrade code, version etc
<Chain>
  <!-- Our BA needs .NET 4.8, so make sure to include that as a prereq! -->
  <PackageGroupRef Id="NetFx48RedistAsPrereq" />
  <MsiPackage .... />
</Chain>
```

**Reply by sampathy10** [reply]
Thank you for the response. When msi package (actual application) has various dialogs that are not displayed using bundle. Is there a solution to keep the actual application dialogs? like custom features, installation folder dialogs etc. 
or 
Can msi include .net runtime install as pre-requisite so that i can retain msi dialogs?  Sample code using wix 5 will be helpful.

**Reply by bevanweiss** [reply]
You’ll want to look under the Bundle and MsiPackage schema.
But yes it can.
